### PR TITLE
GCC: CMake options for unroll-loops and forcing static builds.

### PR DESCRIPTION
**Pull request by TomArrow**
CMake options for compiling on Linux that might be useful to some.

- CompileWithGCCUnrollLoops lets gcc compile with the -funroll-loops option, which unrolls loops more than -O3 already does, potentially improving performance while increasing binary size.
- CompileWithStaticGCCAndSTDCPP lets gcc compile with static libgcc and libstdc++ which can be useful to reduce dependencies to some degree.

Not sure if I did this right, don't know that much about CMAKE or its coding conventions, but seems to work. I'll improve it if needed.

### CODE REVIEW

**Comment by taysta** [comment]
Hi Tom,

Thanks for making the PR, I would like more context before I consider merging it. 

From my understanding, the increased cache size associated with unrolling loops can actually be detrimental to performance, rather than improving it.

Were you able to verify any noticeable differences by a benchmark or something of the sorts? I just wouldn't want to support the claim that people can enable this option for larger, but faster binaries without verifying it somehow.

And for the usage of static libgcc and stdcpp, can you describe the use case for this? I am not too familiar with these options and do not really understand when they may be favourable. 

Thanks

**Comment by TomArrow** [comment]
Yeah it's not guaranteed to improve performance from what I understand and can even be detrimental, hence why I just added it as an option that is off by default. Server owners could then do their own tests with their own use cases and decide whether it's a benefit or not. Basically I was asked to do a variant with this for a server engine and having to manually edit the CMakeLists.txt file between compiles is a bit annoying, so I thought to myself that a switch like this might be handy. I have not done any performance comparisons myself, and am not sure how I would do them tbh.

Re static libgcc and stdcpp, basically that's just about not relying on those libs on the system where the executable is actually used, since those may be the wrong version and then that throws errors and it won't run, and server owners may not want to update a system that works for risk of breaking something. The same is true for libc, but making that one compile statically seems impossible or very hard, so this is kind of a compromise.

Basically just 2 options for comfort when compiling. I feel like they are a bit "unimportant" compared to the other ones but I'm not sure how else to do it, as CMake doesn't allow to categorize options by priority or anything afaik.

**Comment by Alereon** [comment]
The changes proposed in this PR appear to be designed for a specific use case. Given the absence of guaranteed performance enhancements, the lack of evidence indicating that loop bottlenecks exist, and considering that this is not performance-critical code, these changes fall outside the scope of this project.

If you wish to include these specific flags, I recommend learning how to invoke CMake with additional compiler flags. This approach eliminates the need to manually edit the CMakeLists.txt file each time the project is configured.

Closing this PR.

**Comment by TomArrow** [comment]
Fair, thanks for the hint.
